drivers/volume/kdmp/kdmp.go

```diff
@@ -296,13 +320,26 @@ func deleteKdmpSnapshot(backup *storkapi.ApplicationBackup) (bool, error) {
 				logrus.Errorf("%v", errMsg)
 				return false, fmt.Errorf(errMsg)
 			}
-			jobName := getGenericCRName(snapshotDeleteJobPrefix, vInfo.Namespace, vInfo.PersistentVolumeClaim)
+			jobName := getGenericCRName(prefixDelete, string(backup.UID), vInfo.PersistentVolumeClaimUID, vInfo.Namespace)
 			_, err = batch.Instance().GetJob(
 				jobName,
 				vInfo.Namespace,
 			)
 			if err != nil && k8serror.IsNotFound(err) {
-				err := dataexport.CreateCredentialsSecret(secretName, backup.Spec.BackupLocation, backup.Namespace, backup.Namespace)
+				// Adding required label for debugging
+				labels := make(map[string]string)
+				labels[applicationBackupCRNameKey] = getValidLabel(backup.Name)
+				labels[applicationBackupCRUIDKey] = getValidLabel(string(backup.UID))
+				labels[pvcNameKey] = getValidLabel(vInfo.PersistentVolumeClaim)
+				labels[pvcUIDKey] = getValidLabel(vInfo.PersistentVolumeClaimUID)
+				// If backup from px-backup, update the backup object details in the label
+				if val, ok := backup.Annotations[pxbackupAnnotationCreateByKey]; ok {
+					if val == pxbackupAnnotationCreateByValue {
+						labels[backupObjectNameKey] = getValidLabel(backup.Annotations[pxbackupObjectNameKey])
+						labels[backupObjectUIDKey] = getValidLabel(backup.Annotations[pxbackupObjectUIDKey])
+					}
+				}
+				err := dataexport.CreateCredentialsSecret(secretName, backup.Spec.BackupLocation, backup.Namespace, backup.Namespace, labels)
```

Comment:
Should the secret name also follow the new naming convention coz here it just uses `backup.Name` which is the CR name?

Reply:
Fixed